### PR TITLE
Add support NXP EVK-MIMXRT124

### PR DESCRIPTION
**Pull request by Hoohaha**
1. Add MIMXRT124
2. Fix flash programming issue on imxrt1020

### CODE REVIEW

File: pyocd/target/builtin/target_MIMXRT1024xxxxx.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        RamRegion(name="ocram",     start=0x20200000, length=0x20000), # 128 KB
+        FlashRegion(name="flexspi", start=0x60000000, length=0x400000, blocksize=0x1000, is_boot_memory=True,
+            algo=FLASH_ALGO_QUADSPI, page_size=0x100),
+        RamRegion(name="semc",      start=0x80000000, length=0x1e00000, is_external=True), # external sdram
+        RamRegion(name="NCACHE_REGION",      start=0x81e00000, length=0x200000, is_external=True)
+    )
+
+    def __init__(self, link):
+        super(MIMXRT1024xxxxx, self).__init__(link, self.memoryMap)
+        # self._svd_location = SVDFile.from_builtin("MIMXRT1024.xml")
```
Comment (flit) [inline pyocd/target/builtin/target_MIMXRT1024xxxxx.py:100]: Is there not an SVD file for the RT1024?

**Review by flit — APPROVED** [review]

**Comment by flit** [comment]
Thanks! I'm more than happy to add. 😄  But should this and the other RT series (we also need to get RT106x added!) have the same reset-halt related code as for the RT1052?

**Comment by Hoohaha** [comment]
Pull request is updated. Added new family `target_imxrt.IMXRT` for common fbp handling in (set/clear)_reset_catch. I did tests in my local they works fine. I will try to enable imxrt106x if have time.

**Comment by flit** [comment]
Awesome, thanks a lot!

**Comment by flit** [comment]
One more thing… on new files could you please set the copyright to yours alone (either NXP or your own name, it's up to you to decide what's right), and on modified files please add your copyright below existing ones. (I'm trying to get a lot better about this! 😄 )

**Comment by Hoohaha** [comment]
Thanks, updated.

File: pyocd/target/builtin/target_MIMXRT1024xxxxx.py (hunk 1 above)
Comment (Hoohaha) [reply pyocd/target/builtin/target_MIMXRT1024xxxxx.py]: added

**Review by flit — APPROVED** [review]

**Pull request merged**